src/ucp/core/ucp_mm.h

```diff
@@ -41,9 +41,25 @@ typedef struct ucp_mem {
  */
 typedef struct ucp_mem_desc {
     ucp_mem_h                     memh;
+    ucp_mem_h                     meta_memh;
+    void                          *ptr;
+    void                          *meta_ptr;
+    ucs_memory_type_t             mem_type;
+    volatile uint32_t             count;
 } ucp_mem_desc_t;
 
 
+/**
+ * Memory pool private data descriptor.
+ */
+typedef struct ucp_rndv_mpool_priv {
+    ucp_worker_h                  worker;
+    ucs_memory_type_t             mem_type;
+    int                           num_frags;
+    size_t                        frag_size;
```

Comment:
worker->context should already have rndv frag_size

Reply:
Wanted to avoid coupling with worker structure.

Reply:
ok